### ISSUE TITLE
Screenshot and Console Filename Collision

### ISSUE THREAD

**Issue by beckenrode** [laravel/dusk#352]
Screenshot and Console files are overwritten if multiple test classes have functions with the same names.

For example:
`class FooTest extends DuskTestCase {
    public function testExample() {}
}`
would produce a screenshot file named `failure-testExample-0.png`

`class BarTest extends DuskTestCase {
    public function testExample() {}
}`
would also produce a screenshot file named `failure-testExample-0.png` causing the first one to be overwritten.

Would it be better to use the `toString()` method instead of the `getName()` method to generate the filename? This would require cleaning the `toString()` result to remove any special characters but it would then provide a unique filename to stop the filename collision.

**Comment by staudenmeir** [comment]
Fixed in #472.

**Comment by InstanceOfMichael** [comment]
For anyone on an older version of laravel or dusk (I'm using `5.5.*` and `^2.0`), just copying this over the old code works:

```
    /**
     * Capture failure screenshots for each browser.
     *
     * @param  \Illuminate\Support\Collection  $browsers
     * @return void
     */
    protected function captureFailuresFor($browsers)
    {
        $browsers->each(function ($browser, $key) {
            $name = str_replace(['\\', ':'], '_', $this->toString());

            $browser->screenshot('failure-'.$name.'-'.$key);
        });
    }

    /**
     * Store the console output for the given browsers.
     *
     * @param  \Illuminate\Support\Collection  $browsers
     * @return void
     */
    protected function storeConsoleLogsFor($browsers)
    {
        $browsers->each(function ($browser, $key) {
            $name = str_replace(['\\', ':'], '_', $this->toString());

            $browser->storeConsoleLog($name.'-'.$key);
        });
    }
```
